zap/src/main/java/org/zaproxy/zap/extension/search/SearchThread.java

```diff
@@ -279,6 +281,33 @@ private void search() {
                             }
                         }
                     }
+                    if (Type.Note.equals(reqType) && !pcc.allMatchesProcessed()) {
+                        String note = message.getNote();
+                        matcher = pattern.matcher(note);
+
+                        if (inverse && !pcc.allMatchesProcessed()) {
+                            if (!matcher.find()) {
+                                notifyMatchFound(currentRecordId, note, message, null, 0, 0);
+                            }
+                        } else {
+                            while (matcher.find()) {
+                                int noteExtractStart =
+                                        Math.max(matcher.start() - NOTE_EXTRACT_INDEX_OFFSET, 0);
+                                int noteExtractEnd =
+                                        Math.min(
+                                                matcher.end() + NOTE_EXTRACT_INDEX_OFFSET,
+                                                note.length() - 1);
```

Comment:
```suggestion
                                                note.length());
```
As it is exclusive.

Reply:
Interesting, I didn't pay attention that `endIndex` was exclusive in Java. Sorry about that. I realise I worked too fast on this PR.

Fixed in fb1cca021.